jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -5074,9 +5074,12 @@ public String descriptorString() {
 		}
 /*[ENDIF] JAVA_SPEC_VERSION >= 15 */
 		if (this.isArray()) {
-			return name;
+		} else if (this.isPrimitiveClass()) {
+			name = "Q" + name + ";";
+		} else {
+			name = "L" + name + ";";
 		}
-		return "L"+ name + ";"; //$NON-NLS-1$ //$NON-NLS-2$
+		return name; //$NON-NLS-1$ //$NON-NLS-2$
```

Comment:
the `//$NON-NLS-1$` comments need to be updated. Should be above with `name = "Q" + name + ";";`

Reply:
and the L version